ui: Navigate to default view if set when home page loaded

grahammcculloch opened the pull request:
Change-type: minor
Signed-off-by: Graham McCulloch <graham@balena.io>

***

Closes #3712 

This PR finishes off some functionality which seemed to be partially implemented. You can now set a particular view as the default view to load when logging in (or loading the home page). The default view has a little 🏠 icon next to it. You can also now 'unset' the default view - to reset it to normal behaviour.

![DefaultView](https://user-images.githubusercontent.com/2925657/81798787-a1001380-953a-11ea-848d-f32e347e6160.gif)

grahammcculloch committed:
ui: Navigate to default view if set when home page loaded

Change-type: minor
Signed-off-by: Graham McCulloch <graham@balena.io>

StefKors reviewed (changes requested):
Looks great @grahammcculloch 
Nit: Could you add a test for this?

grahammcculloch committed:
dev: Improve UI unit test setup utilities

Relates to #3630 - other PRs can make use of this new utility, either by importing the file before any other imports or specifically importing `getWrapper` from the file and using the generated mock store and/or wrapper.

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

grahammcculloch committed:
dev: Add unit tests for Set as Default functionality

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>
